cmd/buildtools/k0s.go

```diff
@@ -124,12 +127,21 @@ func getK0sVersion() (*semver.Version, error) {
 	return semver.MustParse(v), nil
 }
 
-func getCalicoVersion(opts addonComponentOptions) string {
-	// k0s < 1.32 does not work with calico 3.29+ images due to missing rbac
-	// permissions in the manifests that k0s uses to deploy calico.
-	if opts.k0sVersion.LessThan(semver.MustParse("1.32")) {
-		return "3.28"
+func getCalicoTag(opts addonComponentOptions) (string, error) {
+	calicoVersion := getCalicoVersion(opts)
+	constraints := mustParseSemverConstraints(latestPatchConstraint(calicoVersion))
+	tag, err := GetGreatestGitHubTag(opts.ctx, "projectcalico", "calico", constraints)
+	if err != nil {
+		return "", fmt.Errorf("failed to get calico release: %w", err)
+	}
+	return tag, nil
+}
+
+func getCalicoVersion(opts addonComponentOptions) *semver.Version {
+	// k0s versions prior to 1.31 use calico versions < 3.28,
```

Comment:
this is not a change to how things work prior to this pr as we were already using 3.28 for these olders k0s versions